resources/user_config.js

```diff
@@ -39,8 +39,25 @@ let UserConfig = {
 
       // The plugin auto-detects the language, so set this first.
       // If options files want to override it, they can for testing.
-      if (e.detail.language)
-        Options.Language = e.detail.language;
+      if (e.detail.parserLanguage) {
+        Options.ParserLanguage = e.detail.parserLanguage;
+        // Backward compatibility, everything "Language" should be changed to "ParserLanguage"
+        Options.Language = e.detail.parserLanguage;
+      }
+      const supportedLanguage = ['en', 'de', 'fr', 'ja', 'cn', 'ko'];
+      if (e.detail.systemLocale) {
+        Options.SystemLocale = e.detail.systemLocale;
+        Options.ShortLocale = e.detail.systemLocale.substring(0, 2);
+        if (Options.ShortLocale == 'zh')
+          Options.ShortLocale = 'cn';
+        if (!supportedLanguage.includes(Options.ShortLocale))
+          Options.ShortLocale = Options.ParserLanguage;
+      }
+      if (e.detail.displayLanguage) {
```

Comment:
Can you add some code such that if `e.detail.displayLanguage` is null, that `Options.DisplayLanguage` still gets set?

Reply:
I think removing `if` and adding ` || 'en'` would be okay.